src/rospack_backcompat.cpp

```diff
@@ -40,7 +40,8 @@ namespace rospack
 int
 ROSPack::run(int argc, char** argv)
 {
-  rospack::Rospack rp;
+  // allow caching of results between calls (of same process)
+  static rospack::Rospack rp;
```

Comment:
If API or ABI compatibility is desired this can be done by adding an optional argument to the function.

For API compatibility:

    int ROSPack::run(int argc, char** argv, rospack::Rospack * rp = NULL)

For ABI compatibility it needs to be a separate function:

    int ROSPack::run(int argc, char** argv) {
      rospack::Rospack rp;
      return run(argc, argv, &rp);
    }
    int ROSPack::run(int argc, char** argv, rospack::Rospack * rp)

While this change looks like a reasonable change since we can't rule out that some code changes the environment in between two calls this change would modify the behavior. I will bring this up in the next meeting to determine if we want to go forward with a change like this and in for which ROS distro.